src/raster/rasterband.jl

```diff
@@ -165,23 +177,23 @@ end
 Fetch the no data value for this band.
 
 If there is no out of data value, `nothing` will be
-returned instead. The no data value for a band is generally a special marker value
-used to mark pixels that are not valid data. Such pixels should generally
+returned instead. The no data value for a band is generally a special marker
+value used to mark pixels that are not valid data. Such pixels should generally
 not be displayed, nor contribute to analysis operations.
 
 ### Returns
 the nodata value for this band or `nothing`.
 """
-function getnodatavalue(band::AbstractRasterBand)
+function getnodatavalue(band::AbstractRasterBand)::Union{Cint,Nothing}
```

Comment:
This return value doesn't handle floating point nodatavalue `:Union{Cint,Nothing}`

Reply:
Thanks for the catch! Looks like I mistook `nodatavalue` for `hasnodatavalue` 😅  How did you notice it?

I don't yet know of a good way for me to audit whether the return types correspond to their current behavior, and am curious if others have a better way of doing it.

Reply:
I just ran the GeoData.jl test suite with this branch. That's the only fail :)

Whats the reasoning behind fixing the return type? is it mostly documentation/clarity?

Reply:
> Whats the reasoning behind fixing the return type? is it mostly documentation/clarity?

It seems like a bug to be coercing `Float64`s into `Cint`s.

Reply:
Oh sorry I mean generally everywhere in the package you are setting a fixed return type where there wasn't before, I'm wondering what the motivation is for that

Reply:
> Oh sorry I mean generally everywhere in the package you are setting a fixed return type where there wasn't before, I'm wondering what the motivation is for that

Ohh, yeah, it's mostly documentation/clarity. It was also helpful for me to be able to do a search on `::GDAL.` in this repository to find all instances where I have GDAL types in the input or output.

Reply:
I'm also hoping that, in cases like #171, it'll enable the kind of analysis that allows for us to verify that they are fixed (e.g. if **all** functions have return type annotations, and none of the functions will return `::Dataset` or `::RasterBand` or ...)

Reply:
I'm not sure it can be perfect as there is always the chance of them returning a dataset wrapped in something else, like an array wrapper object or a tuple. I've just been handling this in GeoData.jl as well, for netcdf/hdf5 etc.

Reply:
Just to give an update here: I'm changing the function to return `missing` rather than `nothing` in https://github.com/yeesian/ArchGDAL.jl/pull/179/commits/abe94cdea5ff20233fb96d519ac25c87f8b3c9b1 based on the discussion in https://github.com/yeesian/ArchGDAL.jl/issues/192#issuecomment-851678784.

@visr for vis.

Reply:
Leaving this comment thread unresolved for you to go through it. Feel free to let me know if I might have missed anything.